apps/SageAccountWeb/src/components/RegisterAccount1.tsx

```diff
@@ -312,21 +332,28 @@ export const RegisterAccount1 = () => {
                           id="username"
                           name="username"
                           required
-                          onChange={e => setUsername(e.target.value)}
+                          error={!!usernameInvalidReason}
+                          helperText={usernameInvalidReason}
+                          onChange={e => {
+                            setUsername(e.target.value)
+                          }}
                           value={username || ''}
-                          onKeyPress={(e: any) => {
+                          onKeyDown={e => {
                             if (e.key === 'Enter') {
-                              onSignUpWithGoogle(e)
+                              setUsernameInvalidReason(null)
+                              void onSignUpWithGoogle(e)
                             }
                           }}
                         />
                       </StyledFormControl>
                       <Button
                         sx={buttonSx}
                         variant="contained"
-                        onClick={onSignUpWithGoogle}
+                        onClick={e => {
+                          void onSignUpWithGoogle(e)
```

Comment:
Should this also clear the error state before attempted signup (same as clicking Enter)?
setUsernameInvalidReason(null)